homeassistant/components/imap/config_flow.py

```diff
@@ -128,3 +135,64 @@ async def async_step_reauth_confirm(
             ),
             errors=errors,
         )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> OptionsFlow:
+        """Get the options flow for this handler."""
+        return OptionsFlow(config_entry)
+
+
+class OptionsFlow(config_entries.OptionsFlowWithConfigEntry):
+    """Option flow handler."""
+
+    def _async_abort_entries_match(
+        self, match_dict: dict[str, Any] | None
+    ) -> dict[str, str]:
+        """Validate the user input against other config entries."""
+        if match_dict is None:
+            return {}
+
+        errors: dict[str, str] = {}
+        for entry in [
+            entry
+            for entry in self.hass.config_entries.async_entries(DOMAIN)
+            if entry is not self.config_entry
+        ]:
+            if all(
+                item in entry.data.items()
+                for item in match_dict.items()
+            ):
+                errors["base"] = "already_configured"
+                break
+        return errors
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manage the options."""
+        errors: dict[str, str] = self._async_abort_entries_match(
+            {
+                CONF_SERVER: self._config_entry.data[CONF_SERVER],
+                CONF_USERNAME: self._config_entry.data[CONF_USERNAME],
+                CONF_FOLDER: user_input[CONF_FOLDER],
+                CONF_SEARCH: user_input[CONF_SEARCH],
+            }
+            if user_input
+            else None
+        )
+        entry_data: dict[str, Any] = dict(self._config_entry.data)
+        if not errors and user_input is not None:
+            entry_data.update(user_input)
+            errors = await validate_input(entry_data)
+            if not errors:
+                self.hass.config_entries.async_update_entry(
+                    self.config_entry, data=entry_data
+                )
```

Comment:
I think you need to trigger reload:
```suggestion
            self.hass.config_entries.async_update_entry(
                    self.config_entry, data=entry_data
                )
            self.hass.async_create_task(
                self.hass.config_entries.async_reload(self.config_entry.entry_id)
            )
```